src/main/java/Shape.java

```diff
@@ -0,0 +1,13 @@
+import lombok.extern.slf4j.Slf4j;
+
+public sealed interface Shape permits Circle, Polygon{
+    String getArea();
+
+    default String getDiameter() {
+        return switch (this){
+            case Circle c -> String.format("getDiameter() called for %s", this.toString());
```

Comment:
Hmmm... zastanawiam się - czy to się na pewno kompiluje ? (switch z instanceof pattern to jest preview feature w Javie 17 i normalnie w Mavenie nie przejdzie)

Reply:
Ha.... a próbowałaś tę oryginalną wersję zbudować z linii poleceń ? 😄  In terminal veritas jak to mówili starożytni Rzymianie 😉  - program musi się dać budować w taki sposób, IDE to tylko pomoc